engine/runtime/src/main/scala/org/enso/interpreter/instrument/execution/LocationFilter.scala

```diff
@@ -1,20 +1,28 @@
 package org.enso.interpreter.instrument.execution
 
-import com.oracle.truffle.api.source.{Source, SourceSection}
+import com.oracle.truffle.api.source.SourceSection
+import com.oracle.truffle.api.instrumentation.SourceSectionFilter.IndexRange
 import org.enso.compiler.core.IR
 import org.enso.syntax.text.Location
 
 import scala.collection.mutable
+import org.enso.interpreter.runtime.Module
 
 /** Contains instrumentable source locations.
   *
   * @param sections the list of source sections to instrument
   */
-case class LocationFilter(sections: Set[SourceSection]) {
+case class LocationFilter(
+  sections: Set[SourceSection],
+  ranges: Set[IndexRange]
+) {
 
   /** Get the list of source sections to instrument. */
   def getSections: Array[SourceSection] =
     sections.toArray
+
+  def getRanges: Array[IndexRange] =
```

Comment:
The `RuntimeInstrumentationTest` seems to be failing with `IndexRange`s - they are too inclusive.

Reply:
This is [being addressed](https://github.com/enso-org/enso/compare/wip/jtulach/IncrementalUpdates-182323784...JaroslavTulach:jtulach/RewriteLocationFilter) - successfully it seems.

Reply:
Merged in and all CI tests are green.